### PR TITLE
feat(SPV-737) unify search object & filters and move to models package

### PR DESCRIPTION
**Pull request by chris-4chain**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [x] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

**Review by pawellewandowski98 — APPROVED** [review]

File: actions/admin/access_keys.go [hunk 1]
```diff
@@ -14,24 +15,24 @@ import (
 // @Description	Access Keys Search
 // @Tags		Admin
 // @Produce		json
-// @Param		SearchAccessKeys body SearchAccessKeys false "Supports targeted resource searches with filters and metadata, plus options for pagination and sorting to streamline data exploration and analysis"
+// @Param		SearchAccessKeys body filter.SearchAccessKeys false "Supports targeted resource searches with filters and metadata, plus options for pagination and sorting to streamline data exploration and analysis"
```
Comment (wregulski) [inline actions/admin/access_keys.go:18]: Shouldn't it be `filter.AdminSearchAccessKeys`? (In swagger docs)

File: actions/admin/access_keys.go [hunk 2]
```diff
@@ -52,22 +53,22 @@ func (a *Action) accessKeysSearch(c *gin.Context) {
 // @Description	Access Keys Count
 // @Tags		Admin
 // @Produce		json
-// @Param		CountAccessKeys body CountAccessKeys false "Enables filtering of elements to be counted"
+// @Param		CountAccessKeys body filter.CountAccessKeys false "Enables filtering of elements to be counted"
```
Comment (wregulski) [inline actions/admin/access_keys.go:56]: as above - shouldn't we point to the specific structure: `filter.AdminCountAccessKeys`?

File: actions/admin/access_keys.go (hunk 1 above)
Comment (chris-4chain) [reply actions/admin/access_keys.go]: Thanks 💯 
Fixed! 👍

File: actions/admin/access_keys.go (hunk 2 above)
Comment (chris-4chain) [reply actions/admin/access_keys.go]: Yes. fixed 👍

**Review by wregulski — APPROVED** [review]

**Comment by chris-4chain** [comment]
We encouraged unexpected problem with `mAPI` during tests. Most probably, some test(s) depend on external mAPI provider which was down when GH ran tests. 

I'll re-run tests later and check if the problem still exists.
I decided not to fix this issue because we're removing the mAPI support anyway.

**Pull request merged**